### PR TITLE
Add support for ematch/ipset

### PR DESCRIPTION
**Pull request by Vic063**
Please be advised that the modification made to netlink/__init__.py which increments the 'type' value when encoding nested nla arrays might affect other classes that use it.

### CODE REVIEW

File: pyroute2/netlink/__init__.py [hunk 1]
```diff
@@ -1007,12 +1007,15 @@ def encode(self):
...
         # handle the array case
         if self._nla_array:
+            header_type = 1
             for value in self.getvalue():
                 cell = type(self)(data=self.data,
                                   offset=offset,
                                   parent=self)
                 cell._nla_array = False
-                cell['header']['type'] = 1
+                cell['header']['type'] = header_type
```
Comment (svinota) [inline pyroute2/netlink/__init__.py]: A good question, does it break other array usage cases. Give me some time to check it.

**Comment by svinota** [comment]
cool. I merge this tonight as soon as I get the regression test results.

**Comment by svinota** [comment]
yeah, I see your first comment, it's precise about that :) thanks guys, I'm to run the tests and then I merge.

**Comment by svinota** [comment]
So, some issues found:
```
pyroute2 {Vic063-ematch} $ flake8 pyroute2/
pyroute2/netlink/rtnl/tcmsg/cls_basic.py:48:1: F401 'pyroute2.netlink.rtnl.tcmsg.common_ematch.get_ematch_parms' imported but unused
pyroute2/netlink/rtnl/tcmsg/cls_basic.py:90:20: E126 continuation line over-indented for hanging indent
pyroute2/netlink/rtnl/tcmsg/cls_basic.py:93:20: E123 closing bracket does not match indentation of opening bracket's line
pyroute2/netlink/rtnl/tcmsg/ematch_ipset.py:4:1: E302 expected 2 blank lines, found 1
pyroute2/netlink/rtnl/tcmsg/common_ematch.py:4:12: E126 continuation line over-indented for hanging indent
pyroute2/netlink/rtnl/tcmsg/common_ematch.py:14:11: E126 continuation line over-indented for hanging indent
pyroute2/netlink/rtnl/tcmsg/common_ematch.py:17:21: E126 continuation line over-indented for hanging indent
pyroute2/netlink/rtnl/tcmsg/common_ematch.py:27:21: E123 closing bracket does not match indentation of opening bracket's line
pyroute2/netlink/rtnl/tcmsg/common_ematch.py:81:14: F821 undefined name 'xrange'
pyroute2/netlink/rtnl/tcmsg/common_ematch.py:112:74: E703 statement ends with a semicolon
pyroute2/netlink/rtnl/tcmsg/em_ipset.py:1:1: F401 'struct' imported but unused
pyroute2/netlink/rtnl/tcmsg/em_ipset.py:28:16: F821 undefined name 'TCF_EM_REL_VALID'
pyroute2/netlink/rtnl/tcmsg/em_ipset.py:65:18: F821 undefined name 'xrange'

pyroute2 {Vic063-ematch} $ grep -rn TCF_EM_REL_VALID pyroute2/
pyroute2/netlink/rtnl/tcmsg/em_ipset.py:28:        if not TCF_EM_REL_VALID(flags):
```

 1. Please use flake8 on the code
 2. Please don't use `xrange()` — the code must be Python3-compatible as well. I believe that ranges used here are not so big so it would be necessary to use `xrange()` in Python2, so please use `range()`.
 3. Please define or remove `TCF_EM_REL_VALID`

Thanks!

**Comment by Vic063** [comment]
Hi @svinota 

I just pushed a new commit that fixed mistakes you shown. Sorry for that, I am not really comfortable with PR.

**Comment by svinota** [comment]
Regression tests passed, so I merge that. If there will be issues with the array NLA types — we'll fix it later, but for now I see none.

Thanks, great job!

**Pull request merged**